idl/github.com/uber/cadence/shared.thrift

```diff
@@ -1462,3 +1462,91 @@ struct HistoryBranch{
   20: optional string branchID
   30: optional list<HistoryBranchRange>  ancestors
 }
+
+enum BatchOperationType{
+    Terminate
+    RequestCancel
+    Signal
+}
+
+struct BatchSignalRequest{
+    10: optional string signalName
+    20: optional binary signalData
+}
+
+struct BatchTerminateRequest{
+    10: optional bool applyOnChildren
+}
+
+struct BatchCancelRequest{
+    10: optional bool applyOnChildren
+}
+
+struct StartBatchJobRequest{
+    10: optional string domain
+    20: optional SearchAttributes searchAttributes
+    30: optional BatchOperationType operationType
+    40: optional string sourceQuery
+    50: optional string reason
+    60: optional BatchSignalRequest signalRequest
+    70: optional BatchTerminateRequest terminateRequest
+    80: optional BatchCancelRequest cancelRequest
+    90: optional i64 rps
+    100: optional Header header
+    110: optional i32 startToCloseTimeoutSeconds
+    120: optional string requestID
+    130: optional binary identity
+
+}
+
+struct StartBatchJobResponse{
+    10: optional string jobID
+}
+
+struct StopBatchJobRequest{
+    10: optional string domain
+    20: optional string jobID
+}
+
+struct DescribeBatchJobRequest{
+    10: optional string domain,
+    20: optional string jobID
+}
+
+struct DescribeBatchJobResponse{
+    10: optional i64 totalEstimate
+    20: optional i64 successCount
+    30: optional i64 failureCount
+    40: optional WorkflowExecutionCloseStatus status
+    50: optional i32 currentPage
```

Comment:
This doesn't make sense for the user unless it is possible to kill batch job and restart from that page one day later.

Reply:
That's true. I was thinking to provide visibility of progress but looks like we already have it with SuccessCount/FailureCount/TotalEstimate